tests/components/motionblinds_ble/__init__.py

```diff
@@ -1 +1,61 @@
 """Tests for the Motionblinds Bluetooth integration."""
+
+from unittest.mock import patch
+
+from motionblindsble.const import MotionBlindType
+
+from homeassistant.components.motionblinds_ble import async_setup_entry
+from homeassistant.components.motionblinds_ble.const import (
+    CONF_BLIND_TYPE,
+    CONF_LOCAL_NAME,
+    CONF_MAC_CODE,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.service_info.bluetooth import BluetoothServiceInfo
+
+from tests.common import MockConfigEntry
+
+FIXTURE_SERVICE_INFO = BluetoothServiceInfo(
+    name="MOTION_CCCC",
+    address="cc:cc:cc:cc:cc:cc",
+    rssi=-63,
+    service_data={},
+    manufacturer_data={},
+    service_uuids=[],
+    source="local",
+)
+
+
+async def setup_platform(
+    hass: HomeAssistant,
+    platforms: list[Platform],
+    blind_type: MotionBlindType = MotionBlindType.ROLLER,
+) -> tuple[MockConfigEntry, str]:
+    """Mock a fully setup config entry."""
+
+    config_entry = MockConfigEntry(
+        title="mock_title",
+        domain=DOMAIN,
+        unique_id="cc:cc:cc:cc:cc:cc",
+        data={
+            CONF_ADDRESS: "cc:cc:cc:cc:cc:cc",
+            CONF_LOCAL_NAME: "Motionblind CCCC",
+            CONF_MAC_CODE: "CCCC",
+            CONF_BLIND_TYPE: blind_type.name.lower(),
+        },
+    )
+    config_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.motionblinds_ble.PLATFORMS", platforms):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await async_setup_entry(hass, config_entry)
+        await hass.async_block_till_done()
+        assert config_entry.state is ConfigEntryState.LOADED
+
+    return (
+        config_entry,
+        str(config_entry.data[CONF_LOCAL_NAME]).lower().replace(" ", "_"),
+    )
```

Comment:
I have mixed feelings about this function. I think we should better utilize mocks and fixtures and only patch the platforms when we really have a reason to.

Reply:
I guess the platforms patch can be removed, but I saw others use it so I included it as well (maybe for performance benefit?). The config_entry can be made a fixture, anything else?

Reply:
I mean some integrations do it, but there is no real benefit at this point (like I have never seen anybody patch it for performance benefits, and if that was a thing, this integration would not be big enough for that). I would also remove the `assert config_entry.state is LOADED` because otherwise if you want to test a failing set up you need to copy paste everything

Reply:
Removed the assertion and the platform parameter. What would you use mocks or a fixture for here, considering the MockConfigEntry uses the blind_type parameter from the setup_platform function?

Reply:
You can make the blind_type a fixture as well, you can overwrite that per test

Reply:
I don't understand what the advantage of that would be. You suggest having a fixture for the mock config entry with blind_type as parameter? How would you do that?

Reply:
```py
@pytest.fixture()
def blind_type() -> MotionBlindType:
    return MotionBlindType.ROLLER

@pytest.fixture()
def mock_config_entry(blind_type: MotionBlindType) -> MockConfigEntry:
    return MockConfigEntry(
        data={
            CONF_BLIND_TYPE: blind_type.name.lower(),
        }
    )
```
You can then overwrite the blindtype per test via parametrize

Reply:
And then you would include mock_config_entry in the parameters of each test and pass it to setup_integration() explicitly?

> You can then overwrite the blindtype per test via parametrize

How?

Reply:
Yep, or use a fixture that setup the integration. But I personally like the function more as it feels more flexible.

From the top of my head `@pytest.mark.parametrize("blind_type", ["a", "b"])`

Reply:
Since setup_integration returns a tuple I think it is better it remains a tuple indeed.

> From the top of my head @pytest.mark.parametrize("blind_type", ["a", "b"])

Sure but where and how would you use that to overwrite the default value for blind_type? For example this can use the default blind type:

```python
@pytest.mark.parametrize(("select", "args"), [(ATTR_SPEED, MotionSpeedLevel.HIGH)])
async def test_select(mock_config_entry, hass: HomeAssistant, select: str, args: Any) -> None:
    """Test select."""

    _, name = await setup_integration(hass, mock_config_entry)
```

But test_sensor requires a MotionBlindType.VENETIAN. How can I override the default value of blind_type that is used for mock_config_entry? Previously I just passed it as a kwarg: blind_type=MotionBlindType.CURTAIN

```python
# Parametrize here???
async def test_sensor(
    mock_config_entry,
    hass: HomeAssistant,
    sensor: str,
    update_func: Callable[[MotionDevice], Callable[..., None]],
    initial_value: str,
    args: list[Any],
    expected_value: str,
) -> None:
    """Test sensors."""

    config_entry, name = await setup_integration(
        hass, mock_config_entry # blind_type=MotionBlindType.CURTAIN
    )
    device: MotionDevice = hass.data[DOMAIN][config_entry.entry_id]
```

Reply:
You can parametrize that test yes. You can als parametetrize multiple times on the same test

Reply:
Got it working by putting another parametrize above it